### PR TITLE
Applying CI

### PR DESCRIPTION
**Pull request by skeating**
## Description

This is basic CI. 
There are fixes that need to be done that may be addressed in other PRs.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated all documentation necessary.
- [ ] I have checked spelling in (new) comments.

## Testing
- [X] Testing is done automatically and codecov shows test coverage
- [ ] This cannot be tested automatically

### CODE REVIEW

File: .github/workflows/brief.yml [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    if: github.event.pull_request.draft == false # avoid triggering on draft
+    name:
+      ${{ matrix.platform }}, Parser option ${{ matrix.xml_parser_option }},
+      with namespaces ${{ matrix.with_namespace}}, strict includes ${{
+      matrix.strict }}, with examples ${{ matrix.with_examples}}, package option
+      ${{ matrix.package_option}}
+    strategy:
+      fail-fast: false
+      matrix:
+        xml_parser_option: ["-DWITH_LIBXML", "-DWITH_EXPAT", "-DWITH_XERCES"]
```
Comment (mattagape) [inline .github/workflows/brief.yml:19]: To get the Expat and Xerces options building requires also to have `-DWITH_LIBXML=FALSE`, as LibXML is "on" by default and needs to be "switched off".

File: .github/workflows/brief.yml [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        with:
+          path: ${{runner.workspace}}/.ccache
+          key:
+            ${{ runner.os }}-test-${{ matrix.xml_parser_option }}-${{
+            matrix.with_namespace }}-${{ matrix.strict }}-${{
+            steps.ccache_cache_timestamp.outputs.timestamp }}
+          restore-keys: |
+            ${{ runner.os }}-test-${{ matrix.xml_parser_option }}-${{ matrix.with_namespace }}-${{ matrix.strict }}-${{ steps.ccache_cache_timestamp.outputs.timestamp }}
+            ${{ runner.os }}-test-${{ matrix.xml_parser_option }}-${{ matrix.with_namespace }}-${{ matrix.strict }}-
+            ${{ runner.os }}-test-${{ matrix.xml_parser_option }}-${{ matrix.with_namespace }}-${{ matrix.strict }}-
```
Comment (mattagape) [inline .github/workflows/brief.yml]: duplicated lines?

**Review by mattagape — CHANGES_REQUESTED** [review]
A couple of things to look at.
Most of this stuff I don't understand, so am taking it on faith that it is correct(ish)!!

File: .github/workflows/brief.yml (hunk 1 above)
Comment (fbergmann) [reply .github/workflows/brief.yml:19]: yes, that is checked in the rules below and will be added

File: .github/workflows/brief.yml (hunk 2 above)
Comment (fbergmann) [reply .github/workflows/brief.yml]: those could be removed (specificaly 108-109 i believe), but it works independent of it.

**Review by mattagape — APPROVED** [review]

**Pull request merged**